src/ethereum_test_forks/forks/forks.py

```diff
@@ -291,6 +300,21 @@ def header_beacon_root_required(cls, block_number: int = 0, timestamp: int = 0)
         """
         return True
 
+    @classmethod
+    def pre_allocation(cls, block_number: int = 0, timestamp: int = 0) -> Mapping:
+        """
+        Cancun requires pre-allocation of the beacon root contract for EIP-4788
+        """
+        new_allocation = {
+            0x0B: {
+                "nonce": 1,
+                "code": "0x3373fffffffffffffffffffffffffffffffffffffffe14604457602036146024575f5f"
+                "fd5b620180005f350680545f35146037575f5ffd5b6201800001545f5260205ff35b426201800042"
+                "06555f3562018000420662018000015500",
+            }
+        }
+        return new_allocation | super(Cancun, cls).pre_allocation(block_number, timestamp)
+
```

Comment:
I'm thinking that if we are deploying the beacon root contract on mainnet before the Cancun hardfork then we could add this to Shanghai?

Using the previous method, some of the `ShanghaiToCancunAt15k` transition tests were failing because we didn't have the beacon root in the pre allocation at genesis. As our first block started at Shanghai the pre allocation didn't include the beacon root, so when we got to Cancun block we were missing it in storage and hence got a state root mismatch

Reply:
I added an explanation of this to the description, and I'll work in a bit on adding transition tests so this is tested :raised_hands: